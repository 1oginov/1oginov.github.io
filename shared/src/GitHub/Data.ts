import { Repo } from './Repo';

export interface Data {
  homepageUrl: string;
  login: string;
  repos: Repo[];
  timestamp: number;
  url: string;
}
