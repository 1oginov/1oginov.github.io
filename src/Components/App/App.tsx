import * as React from 'react';

import SocialPresence from 'Components/SocialPresence';
import WorksList from 'Components/WorksList';
import { OPEN_SOURCE_LIBRARIES, OPEN_SOURCE_PROJECTS } from 'Constants';

const App: React.FunctionComponent = () => (
  <>
    <h1>Hi</h1>
    <p>I am Danila, nice to meet you!</p>
    <>
      <h2>Social Presence</h2>
      <SocialPresence />
    </>
    <>
      <h2>Open Source</h2>
      <>
        <h3>Projects</h3>
        <WorksList works={OPEN_SOURCE_PROJECTS} />
      </>
      <>
        <h3>Libraries</h3>
        <WorksList works={OPEN_SOURCE_LIBRARIES} />
      </>
    </>
  </>
);

export default App;
