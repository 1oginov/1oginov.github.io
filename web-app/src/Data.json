{
  "socialPresenceItems": [
    {
      "title": "LinkedIn",
      "url": "https://www.linkedin.com/in/loginov-rocks/"
    },
    {
      "title": "GitHub",
      "url": "https://github.com/loginov-rocks"
    },
    {
      "title": "Medium",
      "url": "https://loginov-rocks.medium.com/"
    },
    {
      "title": "DRIVE2",
      "url": "https://www.drive2.com/r/toyota/chaser/288230376151952785/",
      "url2": "https://www.drive2.ru/r/toyota/chaser/288230376151952785/"
    },
    {
      "title": "Instagram",
      "url": "https://www.instagram.com/loginov_rocks/"
    },
    {
      "title": "Facebook",
      "url": "https://www.facebook.com/1oginov"
    },
    {
      "title": "Twitter",
      "url": "https://twitter.com/loginov_rocks"
    },
    {
      "title": "npm",
      "url": "https://www.npmjs.com/~loginov-rocks"
    }
  ]
}
