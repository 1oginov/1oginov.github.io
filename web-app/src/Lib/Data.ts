import { SocialPresenceItem } from './SocialPresenceItem';

export interface Data {
  socialPresenceItems: SocialPresenceItem[];
}
