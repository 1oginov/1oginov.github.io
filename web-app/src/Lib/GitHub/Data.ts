import { Repo } from './Repo';

export interface Data {
  homepageUrl: string;
  repos: Repo[];
  timestamp: number;
  url: string;
  user: string;
}
