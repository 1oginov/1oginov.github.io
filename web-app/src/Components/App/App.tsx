import { GitHubData as Data } from '@loginov-rocks/loginov-rocks-shared';
import * as React from 'react';

import { SocialPresence } from 'Components/SocialPresence';
import { WorksList } from 'Components/WorksList';
import { OPEN_SOURCE_LIBRARIES, OPEN_SOURCE_PROJECTS, WEB_APP_S3_GITHUB_FILE_KEY } from 'Constants';
import { GitHubData } from 'Lib/GitHubData';

const gitHubData = new GitHubData({ url: `/${WEB_APP_S3_GITHUB_FILE_KEY}` });

export const App: React.FunctionComponent = () => {
  const [data, setData] = React.useState<Data | null>(null);

  React.useEffect(() => {
    gitHubData.get()
      .then((_data) => {
        setData(_data);
      });
  }, []);

  return (
    <>
      <h1>Hi</h1>
      <p>I am Danila, nice to meet you!</p>
      <>
        <h2>Social Presence</h2>
        <SocialPresence />
      </>
      <>
        <h2>Open Source</h2>
        <>
          <h3>Projects</h3>
          <WorksList gitHubRepos={data ? data.repos : []} works={OPEN_SOURCE_PROJECTS} />
        </>
        <>
          <h3>Libraries</h3>
          <WorksList gitHubRepos={data ? data.repos : []} works={OPEN_SOURCE_LIBRARIES} />
        </>
      </>
    </>
  );
};
