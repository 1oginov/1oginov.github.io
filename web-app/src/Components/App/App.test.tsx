/* eslint-disable import/no-extraneous-dependencies */

import { shallow } from 'enzyme';
import * as React from 'react';

import * as data from 'Data.json';

import { App } from './App';

it('matches snapshot', () => {
  const wrapper = shallow(<App data={data} />);

  expect(wrapper).toMatchSnapshot();
});
