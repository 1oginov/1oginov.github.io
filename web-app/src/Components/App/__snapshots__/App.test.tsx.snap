// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Fragment>
  <h1>
    Hi
  </h1>
  <p>
    I am Danila, nice to meet you!
  </p>
  <h2>
    Social Presence
  </h2>
  <SocialPresence />
  <h2>
    Open Source
  </h2>
  <h3>
    Projects
  </h3>
  <WorksList
    gitHubRepos={Array []}
    works={
      Array [
        "Web-Bluetooth-Terminal",
        "Portfolio",
        "Portfolio-Firebase",
        "Firebase-Authenticator",
        "IoT-Firebase-Authentication",
        "Angular-Gulp-Boilerplate",
        "Arduino-Coffeemaker",
        "MeArm-Controller",
        "MQTT-Home-Monitor",
        "NodeMCU-Arduino-PlatformIO",
        "Cordova-Bluetooth-Terminal",
      ]
    }
  />
  <h3>
    Libraries
  </h3>
  <WorksList
    gitHubRepos={Array []}
    works={
      Array [
        "UbxGps",
        "bluetooth-terminal",
        "canvas-heightmap",
        "comments-extractor",
        "gulpdoc",
        "gulp-locales-bundler",
        "locales-bundler",
        "react-native-redux-form",
        "redux-repository",
        "three-sky",
        "ooyala-player-component",
      ]
    }
  />
</Fragment>
`;
